microprofile/telemetry-smallrye/extension/src/main/java/org/wildfly/extension/microprofile/telemetry/MicroProfileTelemetrySubsystemAdd.java

```diff
@@ -36,11 +33,8 @@ protected void performBoottime(OperationContext context, ModelNode operation, Mo
 
         super.performBoottime(context, operation, model);
 
-        ServiceInstaller.builder(ServiceDependency.on(WildFlyOpenTelemetryConfig.SERVICE_DESCRIPTOR))
-                .withCaptor(openTelemetryConfig::set)
-                .build()
-                .install(context);
-
+        final ServiceDependency<WildFlyOpenTelemetryConfig> configServiceDependency =
+                ServiceDependency.on(ServiceName.parse(WildFlyOpenTelemetryConfig.SERVICE_DESCRIPTOR.getName()));
```

Comment:
A single ServiceDependency instance is not safe for use by multiple services.
Consider the following:
1. Deploy application1
   1. Starting the DEPENDENCIES phase service will inject the WildFlyOpenTelemetryConfig into the ServiceDependency to be available to subsequent phases, as expected.
   2. The WildFlyOpenTelemetryConfig can be read from the ServiceDependency in the POST_MODULE phase, as expected.
1. Initiate deploy of application2 followed by undeploy of application1
   1. Starting the DEPENDENCIES phase service (for application2) will inject the WildFlyOpenTelemetryConfig into the already populated ServiceDependency.
   2. Stopping the DEPENDENCIES phase service (for application1) will uninject the WildFlyOpenTelemetryConfig from the ServiceDependency.
   3. If (1) happens before (2), during the POST_MODULE phase of application2, MicroProfileTelemetryDeploymentProcessor can read a null from the shared ServiceDependency instance.